python/tvm/contrib/hexagon/session.py

```diff
@@ -95,6 +94,25 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_value, exc_traceback):
         pass
 
+    @property
+    def device(self):
+        """Session device."""
+
+        if hasattr(self, "_device") and self._device is not None:
```

Comment:
I'd recommend either removing the `hasattr(self, "_device")` from this condition, or removing `self._device = None` from the initializer.  As it is, there are two different ways to represent an uninitialized session, which could cause confusion in the future.